### DISCUSSION TITLE
I have a small question, who can help me improve website performance? Thank you very much

### DISCUSSION THREAD

**Discussion by RyanHo97** [vuejs/vitepress#2378] [General]
I am currently using a flexsearch plugin, and the logo image has been hidden by me. However, I still loaded this resource. To improve performance, what is the method I have to prevent it from being loaded

**Comment by RyanHo97** [comment]
![image](https://github.com/vuejs/vitepress/assets/49436413/78cef075-a75f-49f8-9015-006cb54dc35a)
![image](https://github.com/vuejs/vitepress/assets/49436413/76c683f3-34ab-495e-b626-6df7d9bf6fcc)

**Comment by brc-dd** [comment]
Hiding it via CSS won't work. You need to change it at plugin level 👀  Maybe ask it's authors to provide some option to remove that branding or optimize that svg? (or maybe patch that plugin yourself: https://pnpm.io/cli/patch or https://github.com/ds300/patch-package)

PS: can be hacked via some vite plugin too IG, you'll need to replace that svg part via `transform` (https://vitejs.dev/guide/api-plugin.html, https://rollupjs.org/plugin-development/#transform)

Also, have you tried our [local search](https://vitepress.dev/reference/default-theme-search#local-search)?

**Reply by RyanHo97** [reply]
Thank you for your reply. Your suggestion is very good and cannot be hidden through CSS. I think it may be necessary to configure the plugin to complete the function of not loading SVG.

Yes, I tried your local search, but I didn't want to apply for a key, so I replaced it with another plugin called flexsearch.

**Reply by brc-dd** [reply]
Local search does not require any key 👀  Using Algolia DocSearch requires a key.

**Reply by RyanHo97** [reply]
> Local search does not require any key 👀 Using Algolia DocSearch requires a key.

Is there a built-in local search method besides Algolia DocSearch? If so, please send me a document link. Thank you. If you are busy, I can find it myself.

**Reply by brc-dd** [reply]
Yeah there is. I had linked it above. Pasting it again for reference: https://vitepress.dev/reference/default-theme-search#local-search

**Reply by RyanHo97** [reply]
> Yeah there is. I had linked it above. Pasting it again for reference: https://vitepress.dev/reference/default-theme-search#local-search

Thank you for your help. I have now used local search and the performance is very good🚀

**Comment by RyanHo97** [comment]
Although this method is a bit low-level, I did solve it this way and completed this optimization by changing the dist directory file.

![image](https://github.com/vuejs/vitepress/assets/49436413/300d1a74-4673-4829-a29c-cdb3431af619)

this score:
![image](https://github.com/vuejs/vitepress/assets/49436413/bc395765-5fdf-45a4-8a9c-0bdc579d98dc)
